File: fetch_urls.py
import urllib.request
import re
import json

results = {}

try:
    req = urllib.request.Request('https://gall.dcinside.com/board/lists/?id=neostock', headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)'})
    html = urllib.request.urlopen(req).read().decode('utf-8')
    ids = list(set(re.findall(r'href="/board/view/\?id=neostock&amp;no=(\d+)', html)))
    results['dc_stock'] = ['https://gall.dcinside.com/board/view/?id=neostock&no=' + i for i in ids[:3]]
except Exception as e:
    results['dc_stock_err'] = str(e)

try:
    req = urllib.request.Request('https://gall.dcinside.com/board/lists/?id=immovables', headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)'})
    html = urllib.request.urlopen(req).read().decode('utf-8')
    ids = list(set(re.findall(r'href="/board/view/\?id=immovables&amp;no=(\d+)', html)))
    results['dc_realestate'] = ['https://gall.dcinside.com/board/view/?id=immovables&no=' + i for i in ids[:3]]
except Exception as e:
    results['dc_realestate_err'] = str(e)

try:
    req = urllib.request.Request('https://finance.naver.com/item/board.naver?code=005930', headers={'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64)'})
    html = urllib.request.urlopen(req).read().decode('euc-kr')
    ids = list(set(re.findall(r'board_read\.naver\?code=005930&amp;nid=(\d+)', html)))
    results['naver_stock'] = ['https://finance.naver.com/item/board_read.naver?code=005930&nid=' + i for i in ids[:3]]
except Exception as e:
    results['naver_stock_err'] = str(e)

print(json.dumps(results, indent=2))

File: fetch_urls.js
const https = require('https');

function fetchHtml(url) {
    return new Promise((resolve, reject) => {
        https.get(url, { headers: { 'User-Agent': 'Mozilla/5.0' } }, (res) => {
            let data = '';
            res.on('data', chunk => data += chunk);
            res.on('end', () => resolve(data));
        }).on('error', reject);
    });
}

async function getUrls() {
    try {
        // Naver Stock (Samsung)
        const naverHtml = await fetchHtml('https://finance.naver.com/item/board.naver?code=005930');
        const naverMatch = naverHtml.match(/board_read\.naver\?code=005930&nid=\d+&st=&sw=&page=1/g);
        console.log('Naver:', naverMatch ? `https://finance.naver.com/item/${naverMatch[0]}` : 'not found');

        // DC Inside Neostock
        const dcHtml = await fetchHtml('https://gall.dcinside.com/board/lists/?id=neostock');
        const dcMatch = dcHtml.match(/\/board\/view\/\?id=neostock&no=\d+/g);
        console.log('DC:', dcMatch ? `https://gall.dcinside.com${dcMatch[0]}` : 'not found');

        // Instiz
        const instizHtml = await fetchHtml('https://www.instiz.net/pt');
        const instizMatch = instizHtml.match(/\/pt\/\d+/g);
        console.log('Instiz:', instizMatch ? `https://www.instiz.net${instizMatch[0]}` : 'not found');

    } catch (e) {
        console.error(e);
    }
}

getUrls();
